### PR TITLE
Update MUD tests

### PR DESCRIPTION
**Pull request by didovesei**

### CODE REVIEW

File: testing/test_base.sh [hunk 1]
```diff
@@ -76,28 +76,41 @@ if [ -z `which tcpdump` ]; then
     export PATH=/usr/sbin:$PATH
 fi
 
-device_traffic="tcpdump -en -r inst/run-9a02571e8f01/scans/monitor.pcap port 47808"
-device_bcast="$device_traffic and ether broadcast"
-device_ucast="$device_traffic and ether dst 9a:02:57:1e:8f:02"
-device_xcast="$device_traffic and ether dst 9a:02:57:1e:8f:03"
+device_ufr1="$device_traffic and ether src 9a:02:57:1e:8f:03"
```
Comment (grafnu) [inline testing/test_base.sh:79]: Why keep the cntelr_ lines?

Comment (grafnu) [inline testing/test_base.sh:79]: Why do you need this line still?

File: testing/test_base.sh [hunk 2]
```diff
@@ -76,28 +76,41 @@ if [ -z `which tcpdump` ]; then
...
+    device_bcast="$device_traffic and ether broadcast"
+    device_bfr_peer="$device_traffic and ether broadcast and ether src $peer_mac"
+    device_bfr3="$device_traffic and ether boradcast and ether src 9a:02:57:1e:8f:03"
+    device_ufr_peer="$device_traffic and ether src $peer_mac and dst $device_mac"
+    device_ufr3="$device_traffic and ether src 9a:02:57:1e:8f:03 and dst $device_mac"
+    bfr_peer=$($device_bfr_peer | wc -l)
+    bfr3=$($device_bfr3 | wc -l)
+    ufr_peer=$($device_ufr_peer | wc -l)
+    ufr3=$($device_ufr3 | wc -l)
+    echo device $type $(($bfr_peer > 2)) $(($bfr_3 > 2)) $(($ufr_peer > 0)) $(($ufr3 > 0)) | tee -a $TEST_RESULTS
```
Comment (grafnu) [inline testing/test_base.sh:101]: you don't need the $ in front of $bfr_peer -- it's already in "expression" mode so it's redundant.

File: testing/test_base.sh [hunk 3]
```diff
@@ -76,28 +76,41 @@ if [ -z `which tcpdump` ]; then
...
 
+function test_device_traffic {
+    device_num=$1
+    peer_num=$((3-device_num))
+    device_mac=9a:02:57:1e:8f:0$device_num
+    peer_mac=9a:02:57:1e:8f:0$peer_num
+    
+    device_traffic="tcpdump -en -r inst/run-9a02571e8f0$device_num/scans/monitor.pcap port 47808"
+    device_bcast="$device_traffic and ether broadcast"
+    device_bfr_peer="$device_traffic and ether broadcast and ether src $peer_mac"
```
Comment (grafnu) [inline testing/test_base.sh:93]: Be consistent with ordering of src & dst

File: testing/test_base.sh [hunk 4]
```diff
@@ -76,28 +76,41 @@ if [ -z `which tcpdump` ]; then
...
 cntrlr_xcast="$cntrlr_traffic and ether dst 9a:02:57:1e:8f:03"
 
+function test_device_traffic {
+    device_num=$1
+    peer_num=$((3-device_num))
+    device_mac=9a:02:57:1e:8f:0$device_num
+    peer_mac=9a:02:57:1e:8f:0$peer_num
+    
+    device_traffic="tcpdump -en -r inst/run-9a02571e8f0$device_num/scans/monitor.pcap port 47808"
+    device_bcast="$device_traffic and ether broadcast"
```
Comment (grafnu) [inline testing/test_base.sh:92]: what's device_bcast for?

**Review by grafnu — COMMENTED** [review]
Not sure if it was "ready for review" or not, but I had some time so I took a look!

File: testing/test_base.sh (hunk 1 above)
Comment (didovesei) [reply testing/test_base.sh]: Removed

Comment (didovesei) [reply testing/test_base.sh]: Removed

File: testing/test_base.sh (hunk 2 above)
Comment (didovesei) [reply testing/test_base.sh]: Updated

File: testing/test_base.sh (hunk 3 above)
Comment (didovesei) [reply testing/test_base.sh]: Done

File: testing/test_base.sh (hunk 4 above)
Comment (didovesei) [reply testing/test_base.sh]: Removed

File: testing/test_base.sh [hunk 5]
```diff
@@ -76,35 +76,39 @@ if [ -z `which tcpdump` ]; then
...
-cntrlr_xcast="$cntrlr_traffic and ether dst 9a:02:57:1e:8f:03"
+function test_device_traffic {
+    device_num=$1
+    peer_num=$((3-device_num))
+    device_mac=9a:02:57:1e:8f:0$device_num
+    peer_mac=9a:02:57:1e:8f:0$peer_num
+
+    device_traffic="tcpdump -en -r inst/run-9a02571e8f0$device_num/scans/monitor.pcap port 47808"
+    device_bfr_peer="$device_traffic and ether src $peer_mac and ether broadcast"
+    device_bfr3="$device_traffic and ether src 9a:02:57:1e:8f:03 and ether broadcast"
```
Comment (grafnu) [inline testing/test_base.sh:87]: nit: I'd consider extracting the 3rd MAC address to a named variable (assigned right after peer_mac) just for consistency.

**Review by grafnu — APPROVED** [review]

**Pull request merged**